### ISSUE TITLE
Interface members marked `internal` or `protected` are required to be public in implementing types

### ISSUE THREAD

**Issue by thinker227** [dotnet/roslyn#62005]
**Version Used**: 7.0.100-preview.4.22252.9

**Steps to Reproduce**:

1. Create an interface with a property marked `internal` and a property marked `protected`.
2. Create a class implementing the interface by marking both properties with the same accessibility modifiers as specified by the interface.

```cs
interface IInterface {
    internal int A { get; }
    protected int B { get; }
}
class Class : IInterface {
    internal int A { get; }
    protected int B { get; }
}

```

[Demo on Sharplab](https://sharplab.io/#v2:D4SwdgLgpgTgZgQwMZQAQEl2VolqDeAsAFCpmrjQxgIA2FkqAggagOZQQDcqAvieVQAHGAHtoSaABMGEVACFWHbnxL9iAAQBMqAMK0EAZ0OoAXBixVcaIqXKVYNepWZLOPdYJHiokqDJdFfHZ3VWJ1IA)

**Expected Behavior**: The code compiles without any errors and the properties in the implementing class behave normally (eg. `A` is only accessible within the same assembly and `B` is only accessible in derived types).

**Actual Behavior**: Error CS0535 is reported for both properties stating that "'Class' does not implement interface member 'IInterface.<Name>'. 'Class.<Name>' cannot implement an interface member because it is not public.".

**Comment by xparadoxical** [comment]
[same with `private protected` and `protected internal`](https://sharplab.io/#v2:EYLgtghglgdgPrALgUwE4DMIGNkAICS+MKG2eA3gLABQuduSaMEANg8bgIK7m4DmyRAG5cAXxr1cAB1QB7FFhQATdolwAhHv0EjxtejPnJFyFY1TM2SXAGEtA4WIkHUUAG4QU0uQuWrcACL2Ok7UegACAEy2LBAAznG4IAREJJg4PM505pb+3LwOulneRiZmHJoFIXqShr6mqkys/nZVjjUu7p54dcZ+1kFtRWFAA===)

**Comment by AlekseyTs** [comment]
It is not clear what is the basis for these expectations. Language specification clearly states that only public members of a class/struct are considered as candidates for an implicit interface implementation. If interface implementation is not meant to be public, there is a special syntax for that, usually referred to as an explicit interface implementation syntax. I think the current behavior follows the specification.
